src/display/canvas.js

```diff
@@ -1082,11 +1082,19 @@ var CanvasGraphics = (function CanvasGraphicsClosure() {
       var ctx = this.ctx;
       var current = this.current;
       var x = current.x, y = current.y;
+      var xw, yh, origX, origY, pixelSize = this.getSinglePixelWidth();
       for (var i = 0, j = 0, ii = ops.length; i < ii; i++) {
         switch (ops[i] | 0) {
           case OPS.rectangle:
-            x = args[j++];
-            y = args[j++];
+            origX = args[0];
+            origY = args[1];
```

Comment:
Having these be defined as
```javascript
origX = args[j++];
origY = args[j++];
```
and then simply using `origX`/`origY` when defining `x`/`y` just below instead seems more readable (rather than having a bunch of extra `args[j++]` occurrences).

Reply:
Because of the loop, `args[j++]` is not the same as `args[0]`.